test/test_helper_discovery.rb

```diff
@@ -0,0 +1,26 @@
+def set_session_user_with_perms perms
+  perms = perms.collect{|p| FactoryGirl.create(:permission, :name => p) rescue Permission.find_by_name(p) }
```

Comment:
Creating new permissions brings new problems (duplicity), I'd rather stick with FactoryGirl, but it must be saved (all objects must be stored to get this working).